### PR TITLE
fix: reject overflowing multisig signer indexes in gRPC compat decode

### LINKED ISSUE #214: Avoid multisig signer index overflow panic in gRPC compat decode
## Summary
`agglayer-interop-grpc-types` gRPC v1 compat decode for multisig currently computes `entry.index + 1` when sizing the signer vector.

If `entry.index == u32::MAX`, this overflows and panics in debug/test builds instead of returning a typed validation error.

Agglayer currently needs a local precheck only to avoid that panic before delegating to interop conversion.

## Location
- `crates/agglayer-interop-grpc-types/src/compat/v1/aggchain_data/v1_to_types.rs`

## Current Behavior
The code computes:

```rust
let required_len = signatures
    .iter()
    .map(|entry| entry.index + 1)
    .max()
    .unwrap_or(0);
```

This can panic on `u32::MAX + 1`.

## Expected Behavior
- Use `checked_add(1)` or equivalent safe arithmetic.
- Return a typed `Error::invalid_data("Multisig ECDSA signer index overflow")` instead of panicking.

## Why It Matters
Once this is fixed, Agglayer can rely on interop multisig conversion directly without keeping a local overflow prevalidation step.

## Acceptance Criteria
- No panic on `entry.index == u32::MAX`
- Decode returns a typed invalid-data error
- Add a regression test covering the overflow case

### PR DESCRIPTION
**Pull request by Ekleog-Polygon**
## Summary
- reject `u32::MAX` multisig signer indexes in v1 gRPC compat decode with the typed `Multisig ECDSA signer index overflow` error
- keep the decode path from panicking in debug and test builds by using checked arithmetic when sizing the signer vector
- add a regression test that asserts the overflow case returns the exact invalid-data error

## Testing
- `cargo test -p agglayer-interop-grpc-types`

Fixes #214

### CODE REVIEW

File: crates/agglayer-interop-grpc-types/src/compat/v1/aggchain_data/v1_to_types.rs [hunk 1]
```diff
@@ -136,12 +137,14 @@ impl TryFrom<v1::Multisig> for MultisigPayload {
...
-                    .max()
-                    .unwrap_or(0);
+                // Find the maximum key to determine the vector size.
+                let mut required_len = 0u32;
+                for entry in &signatures {
+                    let signer_len = entry.index.checked_add(1).ok_or_else(|| {
+                        Error::invalid_data("Multisig ECDSA signer index overflow".to_string())
+                    })?;
+                    required_len = required_len.max(signer_len);
+                }
```
Comment (Freyskeyd) [inline crates/agglayer-interop-grpc-types/src/compat/v1/aggchain_data/v1_to_types.rs:147]: Could we do something like that?
```rust
let required_len = signatures
	.iter()
	.map(|entry| entry.index.checked_add(1).ok_or_else(|| {
	    Error::invalid_data("Multisig ECDSA signer index overflow".to_string())
	}))
	.max()
	.transpose()?
	.unwrap_or(0);
```

File: crates/agglayer-interop-grpc-types/src/generated/agglayer/interop/types/v1/agglayer.interop.types.v1.rs
Comment (Freyskeyd) [inline crates/agglayer-interop-grpc-types/src/generated/agglayer/interop/types/v1/agglayer.interop.types.v1.rs:1]: This file wasn't up to date?

Comment (Ekleog-Polygon) [reply crates/agglayer-interop-grpc-types/src/generated/agglayer/interop/types/v1/agglayer.interop.types.v1.rs:1]: CI initially failed so I guess no :shrug:

File: crates/agglayer-interop-grpc-types/src/compat/v1/aggchain_data/v1_to_types.rs (hunk 1 above)
Comment (Ekleog-Polygon) [reply crates/agglayer-interop-grpc-types/src/compat/v1/aggchain_data/v1_to_types.rs:147]: I think we can! I'll look at it tomorrow depending on your answer on Slack — I personally don't find `.transpose()` very readable so I don't intuitively reach out to it, but I'll update the PR to that effect if we want to keep it :smile:

Comment (Ekleog-Polygon) [reply crates/agglayer-interop-grpc-types/src/compat/v1/aggchain_data/v1_to_types.rs:147]: I tried this shape, but the blocker is actually `max()`, not `transpose()`: after the `map`, the iterator item is `Result<u32, Error>`, and `Iterator::max()` requires the item type to implement `Ord`. For `Result<u32, Error>`, that would mean `Error: Ord`, which our compat error type does not implement. So this exact version does not compile, and I kept the explicit loop here.

**Review by Freyskeyd — APPROVED** [review]
